### PR TITLE
feat: add profile image upload

### PR DESCRIPTION
**Pull request by Laban254**
## Description

This update introduces a new endpoint for users to upload profile images. The changes include:

-   **Endpoint**: `POST /api/v1/profile/upload-image`
-   **Supported Formats**: JPG and PNG.
-   **Image Processing**: The backend handles resizing and optimization for storage efficiency and optimal display.
-   **Security**: Images are stored securely and managed efficiently.

## Related Issue (Link to issue ticket)
[issue 678](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/678)

## Motivation and Context

This feature is required to allow users to upload and personalize their profiles with images. Handling image uploads efficiently and securely ensures.
## How Has This Been Tested?

The image upload feature was tested by:

-   **Endpoint Testing**: Verified that the endpoint accepts and processes image uploads in JPG and PNG formats.
-   **Image Processing**: Ensured that uploaded images are resized and optimized properly.
-   **Error Handling**: Checked that informative error messages are returned for invalid formats and size limits exceeded.

Testing was performed in a local FastAPI environment with various image formats and sizes.

## Screenshots (if appropriate - Postman, etc):
Screenshot from 2024-08-06 04-27-35

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-    [x]  New feature (non-breaking change which adds functionality)
-    [ ]  Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-    [x]  My code follows the code style of this project.
-    [ ]  My change requires a change to the documentation.
-    [ ]  I have updated the documentation accordingly.
-    [x] I have read the **CONTRIBUTING** document.
-    [x] I have added tests to cover my changes.
-    [x] All new and existing tests passed.

### CODE REVIEW

File: api/v1/routes/profiles.py [hunk 1]
```diff
@@ -104,3 +108,43 @@ async def reactivate_account(request: Request, db: Session = Depends(get_db)):
         status_code=200,
         message="User reactivation successful",
     )
+
+PROFILE_IMAGE_DIR = "static/profile_images"
+
+@profile.post("/upload-image")
+async def upload_profile_image(
+    file: UploadFile = File(...),
+    # current_user: User = Depends(user_service.get_current_user),
```
Comment (joboy-dev) [inline api/v1/routes/profiles.py:117]: Why did you comment this out

File: api/v1/routes/profiles.py [hunk 2]
```diff
@@ -104,3 +108,43 @@ async def reactivate_account(request: Request, db: Session = Depends(get_db)):
...
+PROFILE_IMAGE_DIR = "static/profile_images"
+
+@profile.post("/upload-image")
+async def upload_profile_image(
+    file: UploadFile = File(...),
+    # current_user: User = Depends(user_service.get_current_user),
+    db: Session = Depends(get_db)
+):
+    # user_id = current_user.id
+    user_id = "54677738"
```
Comment (Goketech) [inline api/v1/routes/profiles.py:121]: you didn't update this

Comment (joboy-dev) [inline api/v1/routes/profiles.py:121]: What is this for

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make appropriate changes

File: api/v1/routes/profiles.py [hunk 3]
```diff
@@ -104,3 +108,43 @@ async def reactivate_account(request: Request, db: Session = Depends(get_db)):
...
+        image_url = f"/static/profile_images/{user_id}.jpg"
+
+        # Update user's avatar URL in the database
+        user = db.query(User).filter(User.id == user_id).first()
+        if user:
+            user.avatar_url = image_url
+            db.commit()
+
+        return JSONResponse(status_code=200, content={"message": "Image uploaded successfully", "image_url": image_url})
+
```
Comment (Goketech) [inline api/v1/routes/profiles.py:148]: make db queries in the service folder

**Review by Goketech — CHANGES_REQUESTED** [review]

File: api/v1/routes/profiles.py (hunk 2 above)
Comment (Laban254) [reply api/v1/routes/profiles.py:121]: On it, I forgot to remove it.

**Comment by Laban254** [comment]
@joboy-dev  @Goketech  working on it.

**Pull request closed without merging**